pinot-broker/src/main/java/org/apache/pinot/broker/requesthandler/BaseSingleStageBrokerRequestHandler.java

```diff
@@ -458,6 +458,14 @@ protected BrokerResponse handleRequest(long requestId, String query, @Nullable S
       }
 
       // Validate QPS quota
+      String database = DatabaseUtils.extractDatabaseFromFullyQualifiedTableName(tableName);
+      if (!_queryQuotaManager.acquireDatabase(database)) {
+        String errorMessage =
+            String.format("Request %d: %s exceeds query quota for database: %s", requestId, query, database);
+        LOGGER.info(errorMessage);
+        requestContext.setErrorCode(QueryException.TOO_MANY_REQUESTS_ERROR_CODE);
+        return new BrokerResponseNative(QueryException.getException(QueryException.QUOTA_EXCEEDED_ERROR, errorMessage));
```

Comment:
add a new error type/code DATABASE_QUOTA_EXCEEDED_ERROR?

Reply:
We have the details on which quota exceed in the error message.
Let me know if you still think its better to have a separate error code and I can add one.